src/Interpreters/IndexContextInfo.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <Storages/MergeTree/RangesInDataPart.h>
+#include <Processors/QueryPlan/ReadFromMergeTree.h>
+
+#include <unordered_map>
+#include <memory>
+#include <optional>
+#include <vector>
+#include <unistd.h>
+
+namespace DB
+{
+
+struct PostingsCacheForStore;
+
+class IndexContextInfo
+{
+public:
+    using IndexPostingsCacheForStoreMap = std::unordered_map<String, std::shared_ptr<const PostingsCacheForStore>>;
+
+    struct PartInfo
+    {
+        DataPartPtr data_part;
+        MarkRanges ranges;
+        IndexPostingsCacheForStoreMap postings_cache_for_store_part;
+    };
+
+    const std::shared_ptr<UsefulSkipIndexes> skip_indexes;
+    const std::vector<std::optional<PartInfo>> part_info_vector;
+};
+
+
+using IndexContextInfoPtr = std::shared_ptr<IndexContextInfo>;
```

Comment:
duplicate of https://github.com/ClickHouse/ClickHouse/blob/c8ce2c0c6374fc19e3aba61f3018bf183588e533/src/Interpreters/Context.h#L575, can we reuse it?